src/main/java/com/example/demo/model/Book.java

```diff
@@ -53,7 +55,6 @@ public class Book {
             name = "books_categories",
             joinColumns = @JoinColumn(name = "book_id"),
             inverseJoinColumns = @JoinColumn(name = "category_id"))
-
     @ToString.Exclude
     @EqualsAndHashCode.Exclude
     private Set<Category> categories = new HashSet<>();
```

Comment:
You exclude it from equals() and hashCode() methods, but you didn't override them, so you have basic implementation from Object class. You can either implement them or just remove the annotation
```suggestion
    private Set<Category> categories = new HashSet<>();
```